libraries/CAN_comms/Src/CAN_comms.c

```diff
@@ -0,0 +1,287 @@
+/**
+ * @file CAN_comms.c
+ * 
+ * @brief CAN communications layer library.
+ * This library contains the implementation for the CAN communications layer, which is responsible
+ * for handling the CAN Rx and Tx messages using tasks and queues. This library is designed for the
+ * STM32F1 series and uses the HAL_CAN library and CMSIS_v2 FreeRTOS.
+ * 
+ * @attention See README for details on how to implement this library.
+ */
+
+
+/* Includes */
+#include "CAN_comms.h"
+
+
+/* Private defines */
+#define CAN_RX_TASK_STACK_SIZE 512
+#define CAN_TX_TASK_STACK_SIZE 512
+#define CAN_RX_QUEUE_SIZE 16
+#define CAN_TX_QUEUE_SIZE 16
+#define CAN_RX_STRUCT_SIZE sizeof(CAN_comms_Rx_msg_t)
+#define CAN_TX_STRUCT_SIZE sizeof(CAN_comms_Tx_msg_t)
+#define NUM_CAN_TX_MAILBOXES 3
+
+
+/* Private varibles */
+CAN_comms_config_t CAN_comms_config;
+osMessageQueueId_t CAN_comms_Rx_queue;
+osMessageQueueId_t CAN_comms_Tx_queue;
+osThreadId_t CAN_comms_Rx_task_handle;
+osThreadId_t CAN_comms_Tx_task_handle;
+osSemaphoreId_t CAN_comms_Tx_mailbox_semaphore;
+const osThreadAttr_t CAN_comms_Rx_task_attributes = {
+    .name = "CAN_comms_Rx_task",
+    .stack_size = CAN_RX_TASK_STACK_SIZE,
+    .priority = (osPriority_t) osPriorityLow,
```

Comment:
Good point. @kcgroulx I think this should be high.

Reply:
There is an edge case where the TEL board receives so many messages to the point where the CAN_comms_Rx_task is not able to empty the queue fast enough. If this task has a higher priority than other tasks, then it will stay in the CAN_comms_Rx_task and not switch. This causes an issue if other tasks are doing important things like refreshing the watchdog. That's why I chose to use osPriorityLow.